tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2064,6 +2064,48 @@ def test_on_demand_builder_with_new_spec(self):
             # This proves that we're picking up the new spec version
             assert spec_in_use.spec_version.flags == {"include_environment_tag"}
 
+    def test_on_demand_builder_with_not_event_type_error(self):
+        field = "count()"
+        query = "!event.type:error"
+        spec = OnDemandMetricSpec(field=field, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY)
+        expected_str_hash = "None;{'inner': {'name': 'event.tags.event.type', 'op': 'eq', 'value': 'error'}, 'op': 'not'}"
+        assert spec._query_str_for_hash == expected_str_hash
+
+        query_builder = TimeseriesMetricQueryBuilder(
+            self.params,
+            dataset=Dataset.PerformanceMetrics,
+            interval=3600,
+            query=query,
+            selected_columns=[field],
+            config=QueryBuilderConfig(
+                on_demand_metrics_enabled=True,
+                on_demand_metrics_type=MetricSpecType.DYNAMIC_QUERY,
+            ),
+        )
+        assert query_builder._on_demand_metric_spec_map.get(field, None) == spec
+        assert query_builder.dataset.name == "PerformanceMetrics"
+        assert query_builder.dataset.value == "generic_metrics"
```

Comment:
@k-fish does this test prove that the builder would work with the query?

Reply:
The fact that one falls back w/ the error and the other doesn't is good enough 👍